### PR TITLE
Add functionality to course create

### PR DESCRIPTION
**Pull request by IsharaEkanayaka**

### CODE REVIEW

File: app/Http/Livewire/Backend/CreateCourses.php [hunk 1]
```diff
@@ -3,38 +3,200 @@
...
+            $this->storeCourse();
+            Log::info('Course created successfully.');
+            $this->resetForm();
+            session()->flash('success', 'Course created successfully.');
+        } catch (\Exception $e) {
+            session()->flash('error', 'There was an error creating the course.');
+        }
     }
 
+    public function updatedacademicProgram($value)
```
Comment (NuwanJ) [inline app/Http/Livewire/Backend/CreateCourses.php:128]: better to be `updatedAcademicProgram`

File: database/migrations/2024_08_23_165504_create_semesters_table.php [hunk 2]
```diff
@@ -16,10 +17,10 @@ public function up()
         Schema::create('semesters', function (Blueprint $table) {
             $table->id();
             $table->string('title', 255);
-            $table->integer('version');
-            $table->enum('academic_program', array_keys(\App\Domains\Semester\Models\Semester::ACADEMIC_PROGRAMS));
+            $table->enum('version', array_keys(Semester::getVersions()));
+            $table->enum('academic_program', array_keys(Semester::getAcademicPrograms()));
             $table->text('description')->nullable();
-            $table->string('url', 255)->unique();
+            $table->string('url', 191)->unique();
```
Comment (NuwanJ) [inline database/migrations/2024_08_23_165504_create_semesters_table.php:23]: Any special reason to make this 191 ?

File: database/migrations/2024_08_24_213634_create_courses_and_course_modules_tables.php [hunk 3]
```diff
@@ -3,8 +3,9 @@
 use Illuminate\Database\Migrations\Migration;
 use Illuminate\Database\Schema\Blueprint;
 use Illuminate\Support\Facades\Schema;
+use App\Domains\Course\Models\Course;
 
-class CreateCourseTable extends Migration
+class CreateCoursesAndCourseModulesTables extends Migration
```
Comment (NuwanJ) [inline database/migrations/2024_08_24_213634_create_courses_and_course_modules_tables.php:8]: Usually, the good practice is not to modify the existing migration files. (See, the date is on the file name)

If you need any further changes (Ex: Change of column name, add or remove ant index, foreign key), it is recommended to create a new file.

Remind me to explain why it is important to have it in the next meeting

File: database/migrations/2024_08_23_165504_create_semesters_table.php (hunk 2 above)
Comment (IsharaEkanayaka) [reply database/migrations/2024_08_23_165504_create_semesters_table.php:23]: My database didn't allow that. I found that varchar(255) field length with 'utf8mb4' can exceed 1000 byte limit for unique indexes. so I reduced the character size.

File: database/migrations/2024_08_24_213634_create_courses_and_course_modules_tables.php (hunk 3 above)
Comment (IsharaEkanayaka) [reply database/migrations/2024_08_24_213634_create_courses_and_course_modules_tables.php:8]: I will create a separate migration file for course_modules.

**Comment by NuwanJ** [comment]
It seems some one has changed migration files. Can you take a look at the conflicts and fix them? (You can merge the target branch in VS code and resolve the conflicts in the GUI level)

**Pull request merged**